### PR TITLE
Making Perl API visible on mxnet.io

### PR DESCRIPTION
**Pull request by sergeykolychev**
@mli 
@piiswrong 
Hello guys,
I think that the Perl API is finally ready to join the happy family of other languages on the 
http://mxnet.io
Here is the overview of what is in the pull
1) Multiple bugfixes (including the fix for recent travis test failures)
2) Full set of unit tests for all subsystems
3) Support for RNN
4) Support for Images and RecordIO
5) Multiple examples
6) Reworked documentation
7) Full set of Perl api .md docs to put onto the http://mxnet.io
Thanks!

### CODE REVIEW

File: docs/api/perl/kvstore.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Provides basic operation over multiple devices (GPUs) on a single device.
+
+### Initialization
+
+Let's consider a simple example. It initializes
+a (int, NDArray) pair into the store, and then pulls the value out.
+
+```perl
+    pdl> $kv = mx->kv->create('local')
```
Comment (mli) [inline docs/api/perl/kvstore.md]: suggest to not use indent here

Comment (sergeykolychev) [inline docs/api/perl/kvstore.md]: @mli yes, I believe 'make html' should work, at least I do not see how the files that I added are any different from say the Scala .md pages.
All links work at least on github.
I'll rework all .md pages and remove the indent in a moment.
Thanks.

**Comment by sergeykolychev** [comment]
@mli Can you please explain of what do you mean by the 'dependencies' for making the perl docs.
I am under impression that .md files should just convert with no external dependencies needed.

**Comment by mli** [comment]
converting .md into html is straightforward. but you may want to build the api documents from source codes as well, e.g.

https://github.com/dmlc/mxnet/blob/master/docs/sphinx_util.py#L27

builds the scala api doc

**Comment by sergeykolychev** [comment]
@mli on the topic of the api docs, Perl has https://metacpan.org/release/AI-MXNet for that, and I am linking to metacpan.org from http://mxnet.io from the index.md
All the detailed API docs will be displayed from over there auto generated from the POD snippets I put into the perl source files.

**Comment by sergeykolychev** [comment]
@mli you may want to cancel all intermediate travis builds that were generated by this PR.

**Pull request merged**